v4.0 crash report - NPE at ContributionBoundaryCallback.kt

From google groups forum.

```
APP_VERSION_CODE=1027
APP_VERSION_NAME=4.0.1
ANDROID_VERSION=7.1.2
PHONE_MODEL=LM-X410.F
STACK_TRACE=java.lang.RuntimeException: Unable to start activity ComponentInfo{fr.free.nrw.commons/fr.free.nrw.commons.contributions.MainActivity}: java.lang.NullPointerException: Parameter specified as non-null is null: method kotlin.jvm.internal.Intrinsics.checkNotNullParameter, parameter <set-?>
at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2763)
at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2824)
at android.app.ActivityThread.-wrap12(ActivityThread.java)
at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1546)
at android.os.Handler.dispatchMessage(Handler.java:102)
at android.os.Looper.loop(Looper.java:154)
at android.app.ActivityThread.main(ActivityThread.java:6351)
at java.lang.reflect.Method.invoke(Native Method)
at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:896)
at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:786)
Caused by: java.lang.NullPointerException: Parameter specified as non-null is null: method kotlin.jvm.internal.Intrinsics.checkNotNullParameter, parameter <set-?>
at fr.free.nrw.commons.contributions.ContributionBoundaryCallback.setUserName(ContributionBoundaryCallback.kt)
at fr.free.nrw.commons.contributions.ContributionsListPresenter.setup(ContributionsListPresenter.java:72)
at fr.free.nrw.commons.contributions.ContributionsListFragment.initRecyclerView(ContributionsListFragment.java:190)
at fr.free.nrw.commons.contributions.ContributionsListFragment.onViewCreated(ContributionsListFragment.java:174)
at androidx.fragment.app.FragmentStateManager.createView(FragmentStateManager.java:322)
at androidx.fragment.app.FragmentManager.moveToState(FragmentManager.java:1185)
at androidx.fragment.app.FragmentManager.moveToState(FragmentManager.java:1354)
at androidx.fragment.app.FragmentManager.moveFragmentToExpectedState(FragmentManager.java:1432)
at androidx.fragment.app.FragmentManager.moveToState(FragmentManager.java:1495)
at androidx.fragment.app.FragmentManager.dispatchStateChange(FragmentManager.java:2617)
at androidx.fragment.app.FragmentManager.dispatchActivityCreated(FragmentManager.java:2569)
at androidx.fragment.app.Fragment.performActivityCreated(Fragment.java:2722)
at androidx.fragment.app.FragmentStateManager.activityCreated(FragmentStateManager.java:336)
at androidx.fragment.app.FragmentManager.moveToState(FragmentManager.java:1186)
at androidx.fragment.app.FragmentManager.moveToState(FragmentManager.java:1354)
at androidx.fragment.app.FragmentManager.moveFragmentToExpectedState(FragmentManager.java:1432)
at androidx.fragment.app.FragmentManager.moveToState(FragmentManager.java:1495)
at androidx.fragment.app.BackStackRecord.executeOps(BackStackRecord.java:447)
at androidx.fragment.app.FragmentManager.executeOps(FragmentManager.java:2167)
at androidx.fragment.app.FragmentManager.executeOpsTogether(FragmentManager.java:1990)
at androidx.fragment.app.FragmentManager.removeRedundantOperationsAndExecute(FragmentManager.java:1945)
at androidx.fragment.app.FragmentManager.execPendingActions(FragmentManager.java:1847)
at androidx.fragment.app.FragmentManager.dispatchStateChange(FragmentManager.java:2621)
at androidx.fragment.app.FragmentManager.dispatchActivityCreated(FragmentManager.java:2569)
at androidx.fragment.app.FragmentController.dispatchActivityCreated(FragmentController.java:247)
at androidx.fragment.app.FragmentActivity.onStart(FragmentActivity.java:541)
at androidx.appcompat.app.AppCompatActivity.onStart(AppCompatActivity.java:201)
at android.app.Instrumentation.callActivityOnStart(Instrumentation.java:1249)
at android.app.Activity.performStart(Activity.java:6792)
at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2726)
... 9 more
```

--- a/app/src/main/java/fr/free/nrw/commons/contributions/ContributionBoundaryCallback.kt
+++ b/app/src/main/java/fr/free/nrw/commons/contributions/ContributionBoundaryCallback.kt
@@ -21,7 +21,7 @@ class ContributionBoundaryCallback @Inject constructor(
     @param:Named(CommonsApplicationModule.IO_THREAD) private val ioThreadScheduler: Scheduler
 ) : BoundaryCallback<Contribution>() {
     private val compositeDisposable: CompositeDisposable = CompositeDisposable()
-    lateinit var userName: String
+    var userName: String? = null
 
 
     /**
@@ -53,13 +53,13 @@ class ContributionBoundaryCallback @Inject constructor(
     /**
      * Fetches contributions using the MediaWiki API
      */
-    fun fetchContributions() {
+    private fun fetchContributions() {
         if (sessionManager.userName != null) {
-            compositeDisposable.add(
-                mediaClient.getMediaListForUser(userName!!)
+            userName?.let { userName ->
+                mediaClient.getMediaListForUser(userName)
                     .map { mediaList ->
-                        mediaList.map {
-                            Contribution(media = it, state = Contribution.STATE_COMPLETED)
+                        mediaList.map { media ->
+                            Contribution(media = media, state = Contribution.STATE_COMPLETED)
                         }
                     }
                     .subscribeOn(ioThreadScheduler)
@@ -69,11 +69,13 @@ class ContributionBoundaryCallback @Inject constructor(
                             error.message
                         )
                     }
-            )
-        }else {
-            if (compositeDisposable != null){
-                compositeDisposable.clear()
+            }?.let {
+                compositeDisposable.add(
+                    it
+                )
             }
+        }else {
+            compositeDisposable.clear()
         }
     }
 

